sources/game/Game.gd

```diff
@@ -77,3 +79,32 @@ func end(won: bool) -> void:
 
 func on_level_end() -> void:
 	next()
+
+
+func puzzle_start(scene: PackedScene) -> Node2D:
+	"""
+	Player enters a puzzle scene
+	"""
+	Utils.get_player().disable()
+
+	puzzle = scene.instance()
+	add_child(puzzle)
+
```

Comment:
I think we need update level manager here, or leave a # TODO

Reply:
I don't get your point here, what do you mean?